fix Kusudama pop count parsing and improve big/ex note display and on

IepIweidieng opened the pull request:
Other improvements include the proper support of `#BARLINEOFF`/`#BARLINEON`.

## Test Cases

### [1-hit-kusudamas.tja](https://github.com/user-attachments/files/18297435/1-hit-kusudamas.tja.txt)

:x: Before: Wrong pop count, missing Kusudamas

![All preview](https://github.com/user-attachments/assets/116bad6d-09ec-4929-8c92-c2e8566c1ed5)

:o: After

![All preview (1)](https://github.com/user-attachments/assets/970c9c18-e99b-485f-9f89-c62080f5ce44)

### [rolls-ended-by-nonrolls.tja](https://github.com/user-attachments/files/18297495/rolls-ended-by-nonrolls.tja.txt)

:x: Before: `BALLOON count mismatch`

![{EB5CF091-96DD-4B9E-855A-234F97352CF5}](https://github.com/user-attachments/assets/29ddae8f-b91a-405a-bca6-689774318075)

⚠️ After: Rolls' ends overlap with hit notes. (purely a display issue)

![Rolls Ended by Non-rolls preview](https://github.com/user-attachments/assets/1fa5f4b5-94ab-4916-9f0a-95ea7e95de73)

### [repeat-roll-heads.tja](https://github.com/user-attachments/files/18297530/repeat-roll-heads.tja.txt)

:x: Before: `BALLOON count mismatch`

:o: After

![Repeated Roll Heads preview](https://github.com/user-attachments/assets/57085d0a-d035-4c02-b1ae-8c0400275440)

### [gimmick_test - fixed.tja](https://github.com/user-attachments/files/18297562/gimmick_test.-.fixed.tja.txt)

:x: Before: `Cannot read properties of undefined (reading 'toString')`

:o: After: 

![Gimmick Charting Practice - Fixed preview (1)](https://github.com/user-attachments/assets/6f6316f1-7b7c-4c2a-8230-68f03d5c4344)

WHMHammer commented:
Hi @IepIweidieng. Thank you for your contribution!
- `1-hit-kusudamas.tja`:
    - That is indeed an issue. Thank you for spotting that and fixing it! I am reviewing the code changes. Please allow me with some time.
- In `rolls-ended-by-nonrolls.tja`:
    - Line 14 is `50585058,`. The 1st note (`5`) starts a drum roll. The 4th (`8`) ends it. However, the third note, which is with the drum roll, is also a `5`. I believer this is an undefined behavior, and I'm not sure how it's going to affect the parsing since I'm not the original author of this project. Hence I prefer to have it changed to `50085008` during the rest of our discussion. Please kindly remind me if I am not aware of any rule related to it and point me to the references of the rule.
    - The 1st note in line 17 (`5`) starts another drum roll. However, there is no corresponding `8` to end it. There are a bunch of other notes after the `5`. From the screenshot taken with your changes, I guess you are suggesting that any 1-hit notes after a not-ended-yet drum roll start should still function as itself and also end the drum roll at the same time. Did I get you right? I am not aware of that rule, would you provide me with references? Thanks!
- In `repeat-roll-heads.tja`:
    - For the same reasons as listed above, I believe that lines starting from line 55 contain undefined behaviors, so failure in parsing this file does not need to be fixed.
- Making big note statistics bold is a great idea. The changes related to that also look good to me. Thanks!

IepIweidieng commented:
Quote TaikoJiro 2.92 [readme.txt](https://github.com/user-attachments/files/18341805/readme.txt):
>		連打開始から8までの間を連打とします。「500008」見たいな。
>		「555558」のように同じ数字を敷き詰めても多分大丈夫です。
>		鈴やでんでん等の音符には未対応。
>
>		(Ver.2.36以降）
>		ゲキ連打の終了を省略した場合、風船を割れなかった次の音符は強制的に不可扱いとなります。
>
>		(Ver.2.75以降）
>		イモ連打の開始から終了の間に「9」を入れると、その位置が5000点獲得のボーダーになります。
>		「909900008」のように間に二つ以上9がある場合は、一番後ろの9の位置が使用されます。
>		「900000008」のように間に9を挟まない場合は連打の長さの0.6倍の位置がボーダーとなります。
>		ボーダーよりも早い時間で連打を叩ききると5000点、それ以降は1000点の得点が入ります。

The case `5678` is not mentioned but TaikoJiro handles it as same as `5008`.

Other cases are not explicitly mentioned. I conducted several tests against TaikoJiro and treat the intuitive behaviors as intended and the obscurer behaviors as *unspecified*. See: <https://github.com/IepIweidieng/TJAPlayer3/blob/gh-pages/tja.md#duration-of-drumroll-type-notes>

IepIweidieng commented:
Relative issue(s): https://github.com/kairera0467/TJAP2fPC/issues/74
The new parse behavior has been implemented in OpenTaiko: <https://github.com/0auBSQ/OpenTaiko/pull/756>

IepIweidieng commented:
I had forgotten the unended chart-final roll case. Now fixed. I also updated the `rolls-ended-by-nonrolls.tja` to get rid of the last 1/192 measure for better visualization.

WHMHammer commented:
Sorry for the long wait. I have been very busy recently for some personal affairs. I haven't had the time to fully review the changes yet, but I don't want to block the fixes from being merged, so I will merge them now and review them later. Sorry and thank you!
